pkg/driver/node/credentialprovider/provider_pod.go

```diff
@@ -141,7 +160,53 @@ func (c *Provider) findPodServiceAccountRole(ctx context.Context, provideCtx Pro
 
 // podLevelServiceAccountTokenName returns service account token name for Pod-level identity.
 // It escapes from slashes to make this token name path-safe.
-func podLevelServiceAccountTokenName(podID string, volumeID string) string {
+func podLevelServiceAccountTokenName(podID string, volumeID string) string { // TODO: Consider reusability with podLevelEksPodIdentityServiceAccountTokenName or at least renaming this one.
 	id := escapedVolumeIdentifier(podID, volumeID)
 	return id + ".token"
 }
+
+// podLevelEksPodIdentityServiceAccountTokenName returns service account token name for Pod-level identity with EKS Pod Identity.
+// It escapes from slashes to make this token name path-safe.
+func podLevelEksPodIdentityServiceAccountTokenName(podID string, volumeID string) string {
+	id := escapedVolumeIdentifier(podID, volumeID)
+	return id + "-eks-pod-identity.token"
+}
+
+// createIRSACredentialsEnvironment creates an environment with the environment variables needed for pod-level authentication with IRSA
+func (c *Provider) createIRSACredentialsEnvironment(ctx context.Context, provideCtx ProvideContext) (envprovider.Environment, error) {
+	roleARN, err := c.findPodServiceAccountRole(ctx, provideCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	region, err := c.stsRegion(provideCtx)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Failed to detect STS AWS Region, please explicitly set the AWS Region, see "+stsConfigDocsPage)
+	}
+
+	defaultRegion := os.Getenv(envprovider.EnvDefaultRegion)
+	if defaultRegion == "" {
+		defaultRegion = region
+	}
+
+	tokenName := podLevelServiceAccountTokenName(provideCtx.PodID, provideCtx.VolumeID)
+	tokenFile := filepath.Join(provideCtx.EnvPath, tokenName)
+
+	return envprovider.Environment{
+		envprovider.EnvRoleARN:              roleARN,
+		envprovider.EnvWebIdentityTokenFile: tokenFile,
+		envprovider.EnvRegion:               region,
+		envprovider.EnvDefaultRegion:        defaultRegion,
```

Comment:
Curious why we are passing `region` and `defaultRegion` for IRSA, but not for Pod Identity?

Reply:
Seems like EKS Pod Identity Webhook provides `region` and `defaultRegion` variables for both IRSA and Pod Identity: https://github.com/philljie/amazon-eks-pod-identity-webhook/blob/master/pkg/handler/handler.go#L211-L246

I think we should do the same to be consistent.